utils.d.ts

```diff
@@ -76,3 +76,34 @@ export const asEffect: {
   getContext(effect: Effect): undefined | GetContextEffectDescriptor;
   setContext(effect: Effect): undefined | SetContextEffectDescriptor<any>;
 };
+
+interface SagaIteratorClone extends SagaIterator {
+  clone: () => SagaIteratorClone;
+}
+
+export function cloneableGenerator(
```

Comment:
I think this can be simplified to:

```ts
function cloneableGenerator<S extends (...args: any[]) => SagaIterator>(
  generator: S,
): S & {clone(): ...}
```